nodejs/week3/meal-sharing/src/backend/api/meals.js

```diff
@@ -0,0 +1,106 @@
+const express = require("express");
+const router = express.Router();
+const knex = require("../database");
+
+//...........................................Returns all meals	
+
+router.get("/", async (request, response) => {
+  const {
+    maxPrice,
+    availableReservations,
+    title,
+    createdAfter,
+    limit
+  } = request.query;
+  try {
+    // knex syntax for selecting things. Look up the documentation for knex for further info
+    if (Object.keys(request.query).length === 0) {
+      const resault = await knex("meals").select("*");
+      response.json(resault);
+
+      //.............................................Get meals that has a price smaller than maxPrice	
+
+    } else if (maxPrice) {
+      let resault = await knex("meals").select("*").where("price", "<", maxPrice);
```

Comment:
Good usage of knex/sql instead of doing this kind of logic in the server. Pulling data using SQL will almost always be more efficient than pulling data and then using the server to filter the data.